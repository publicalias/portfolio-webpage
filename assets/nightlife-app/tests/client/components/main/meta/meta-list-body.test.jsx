"use strict";

//local imports

const MetaListBody = require("../../../../../scripts/client/components/main/meta/meta-list-body");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { testMock } = require("redux/tests/meta-tests");
const { initTestEvent, initTestSnapshot, reactTests, withDataList } = require("redux/tests/react-tests");

//node modules

const React = require("react");

//utilities

const { testShallow } = testWrapper(MetaListBody);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(MetaListBody));

//meta list body

describe("meta list body", () => {

  const testList = withDataList(testShallow, [null, {
    handleScroll: jest.fn(),
    header: <div />,
    list: [],
    placeholder: "No Items"
  }]);

  const testSnapshot = initTestSnapshot(testList);

  const testScroll = initTestEvent(testList, "scroll", {});

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (list)", () => testSnapshot(null, { list: [<div key="" />] }));

  it("should call handleScroll on scroll", () => {

    const handleScroll = jest.fn();

    return testScroll(".qa-infinite-scroll", [null, { handleScroll }], () => {
      testMock(handleScroll, [{}]);
    });

  });

});
