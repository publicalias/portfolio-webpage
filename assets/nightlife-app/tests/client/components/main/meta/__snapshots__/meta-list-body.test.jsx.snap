// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta list body should match snapshot (default) 1`] = `
<div
  className="c-list-body"
>
  <div />
  <hr />
  <div
    className="c-list-body__head u-margin-full"
  >
    <h4>
      Photo
    </h4>
    <h4>
      Description
    </h4>
  </div>
  <div
    className="c-list-body__body js-infinite-scroll qa-infinite-scroll"
    onScroll={[MockFunction]}
  >
    <div
      className="c-list-body__content"
    >
      <div>
        <div
          className="c-list-item"
        >
          <p>
            N/A
          </p>
          <p>
            No Items
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`meta list body should match snapshot (list) 1`] = `
<div
  className="c-list-body"
>
  <div />
  <hr />
  <div
    className="c-list-body__head u-margin-full"
  >
    <h4>
      Photo
    </h4>
    <h4>
      Description
    </h4>
  </div>
  <div
    className="c-list-body__body js-infinite-scroll qa-infinite-scroll"
    onScroll={[MockFunction]}
  >
    <div
      className="c-list-body__content"
    >
      <div
        key=""
      />
    </div>
  </div>
</div>
`;
